src/mrsimulator/spin_system/isotope.py

```diff
@@ -21,6 +24,50 @@
 }
 
 
+class IsotopeReferences:
+    _instance = None
+
+    def __new__(cls, *args, **kwargs):
+        if not cls._instance:
+            cls._instance = super().__new__(cls, *args, **kwargs)
+        return cls._instance
+
+    def __init__(self):
+        if not hasattr(self, "initialized"):  # Ensure __init__ is only called once
+            self.initialized = True
+            self.isotopes_data = REFERENCE_DATA
+
+    def get_isotope_data(self, symbol: str):
+        return self.isotopes_data.get(symbol, None)
```

Comment:
This is not necessasy. The REFERENCE_DATA is globally loaded at line 17. Python only executes module level code once during the first import, caches it, and use the cache for subsequent imports.

Reply:
This section of code can be removed